nin/backend/generate/generate.js

```diff
@@ -1,21 +1,66 @@
 var fs = require('fs'),
     path = require('path'),
-    utils = require('../utils');
+    utils = require('../utils'),
+    mkdirp = require('mkdirp');
 
-var generate = function(toGenerate, path) {
-  var projectRoot = utils.findProjectRoot(path);
+var generate = function(type, name) {
+  var projectRoot = utils.findProjectRoot(process.cwd());
   if (projectRoot == '') {
     process.stderr.write('Could not find nin.json in project root\n');
     process.exit(1);
   }
 
-  if (endsWith(toGenerate, 'Layer')) {
-    generateLayer(toGenerate, projectRoot);
-    addToLayers(toGenerate, projectRoot);
+  if (name == "") {
+    return;
+  }
+
+  var camelizedName = utils.camelize(name);
+
+  switch (type) {
+    case 'layer':
+      generateLayer(camelizedName, 'TemplateLayer.js',
+          [[/TemplateLayer/g, camelizedName]],
+          projectRoot);
+
+      addToLayers(camelizedName, 'red', projectRoot);
+      break;
+
+    case 'shader':
+      generateShader(camelizedName, projectRoot);
+      break;
+
+    case 'shaderWithLayer':
+      var shaderLayerName = camelizedName + 'Layer';
+
+      generateShader(camelizedName, projectRoot);
+      generateLayer(shaderLayerName, 'TemplateShaderLayer.js',
+          [[/TemplateLayer/g, shaderLayerName],
+           [/TemplateShader/g, camelizedName]],
+          projectRoot);
+
+      addToLayers(camelizedName + 'Layer', 'pink', projectRoot);
```

Comment:
Yes. This is only called if you generate a Shader + Layer, where you enter the shader name, so if you enter the shader name MyShader in the frontend, it will generate a shader with name MyShader, and a shaderLayer with the name MyShaderLayer.